src/CheckedFile.cpp

```diff
@@ -384,25 +437,38 @@ void CheckedFile::seek(uint64_t offset, OffsetMode omode)
 
 uint64_t CheckedFile::lseek64(int64_t offset, int whence)
 {
+   int64_t result = -1;
+   if(fd_ >= 0){
 #if defined(_WIN32)
 #  if defined(_MSC_VER) || defined(__MINGW32__) //<rs 2010-06-16> mingw _is_ WIN32!
-   __int64 result = _lseeki64(fd_, offset, whence);
+   result = _lseeki64(fd_, offset, whence);
 #  elif defined(__GNUC__) //<rs 2010-06-16> this most likely will not get triggered (cygwin != WIN32)?
 #    ifdef E57_MAX_DEBUG
    if (sizeof(off_t) != sizeof(offset))
       throw E57_EXCEPTION2(E57_ERROR_INTERNAL, "sizeof(off_t)=" + toString(sizeof(off_t)));
 #    endif
-   int64_t result = ::lseek(fd_, offset, whence);
+   result = ::lseek(fd_, offset, whence);
 #  else
 #    error "no supported compiler defined"
 #  endif
 #elif defined(__linux__)
-   int64_t result = ::lseek64(fd_, offset, whence);
+   result = ::lseek64(fd_, offset, whence);
 #elif defined(__APPLE__)
-   int64_t result = ::lseek(fd_, offset, whence);
+   result = ::lseek(fd_, offset, whence);
 #else
 #  error "no supported OS platform defined"
 #endif
+   }
+   else if(bufView_ != nullptr){
+      if(bufView_->seek(offset, whence)){
+         result = bufView_->pos();
+      }
+      else{
+         cerr << "CheckedFile::lseek64 Error: cursorStream_ is out of bound!" << endl;

```

Comment:
Please remove.

This already throws an exception (next block). None of the code in this file is outputting any errors on `std::cerr`.